Classes/Domain/Model/TranslationFile.php

```diff
@@ -13,25 +14,32 @@
 
 use TYPO3\CMS\Core\Core\Environment;
 use TYPO3\CMS\Core\Localization\LocalizationFactory;
+use TYPO3\CMS\Core\Package\PackageManager;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
 
 class TranslationFile extends AbstractTranslationFile
 {
-    /**
-     * @var L10nTranslationFile[]
-     */
+    /** @var L10nTranslationFile[] */
     protected array $l10nTranslationFiles = [];
 
-    /**
-     * @param \SplFileInfo $splFileInfo
-     * @param array $languages
-     * @param LocalizationFactory $localizationFactory
-     */
     public function initFileSystem(\SplFileInfo $splFileInfo, array $languages, LocalizationFactory $localizationFactory): void
     {
+        /** @var PackageManager $packageManager */
+        $packageManager = GeneralUtility::makeInstance(PackageManager::class);
         $this->splFileInfo = $splFileInfo;
-
-        $pathPart = str_replace('/', '\/', Environment::getExtensionsPath() . DIRECTORY_SEPARATOR);
+        // Escape directory separators in path to vendor directory
+        $pathPart = str_replace('/', '\/', Environment::getProjectPath() . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR);
+        // Remove the path to the vendor directory from the path to the file
         $this->relativePath = preg_replace('/' . $pathPart . '/', '', $this->getCleanPath());
+        // Find the vendor/package-name used by composer
+        $matches = [];
+        preg_match('#^[^\/]*\/[^\/]*#', $this->relativePath, $matches);
+        // Convert the composer package name into the TYPO3 extension name.
+        // Append the extension name instead of the composer package name to the relative path.
+        // Example: Instead of "vendor-name/my-extension/Resources/Private/" we will get "my_extension/Resources/Private/",
```

Comment:
what will happen if package key is foo/barBaz ? and extensionKey is bar_baz?
cant we get die EXT-Key from `Package->getComposerManifest('vendor-name/my-extension')`?

Reply:
that would still work